### ISSUE TITLE
consider moving the japanese language support to the blevex repo

### ISSUE THREAD

**Issue by mschoch** [blevesearch/bleve#341]
About once a month someone complains that 'go get github.com/blevesearch/bleve' hangs forever.  When debugging this, its always stuck on:

github.com/ikawaha/kagome

The repo isn't really that large, but for whatever reason it seems to cause lots of problems.

**Comment by otoolep** [comment]
+1, this got me too.

**Comment by dc0d** [comment]
+1, me too; and it makes it look as if something is broken.

**Comment by Shugyousha** [comment]
Hi Marty

Yes, I think that makes sense. People who need the Japanese analyzer
should be able to find it (maybe it could be mentioned more
prominently that more analyzers can be found in blevex?).

The reason why it takes so long is that kagome includes more than
200MB of binary data in github.com/ikawaha/kagome/internal/dic/data.
The data is gzipped and then put into go files as byte strings which
results in the compilation taking a long time it seems.


Cheers,

Silvan

**Comment by ikawaha** [comment]
Recently, it has added support for new dictionary (UniDic) to kagome.
Kagome includes tokenize dictionaries IPADic and UniDic.
So repository may be bit large to go get.

I created IPADic support only version.  
It was reduced repository size.
Please try it.

https://github.com/ikawaha/kagome.ipadic/

**Comment by Shugyousha** [comment]
On Thu, Feb 18, 2016 at 07:27:57PM -0800, ikawaha wrote:
> Recently, it has added support for new dictionary (UniDic) to kagome.
> Kagome includes tokenize dictionaries IPADic and UniDic.
> So repository may be bit large to go get.
> 
> I created IPADic support only version.  
> It was reduced repository size.
> Please try it.
> 
> https://github.com/ikawaha/kagome.ipadic/

Thanks for doing this!

I have tested importing both versions and go getting them (after after
deleting my local copies). The following are the results (on my 8GB RAM
i7 machine).


kagome.ipadic
-------------
time go get github.com/blevesearch/bleve/...

real    0m13.999s
user    0m41.937s
sys     0m3.107s

kagome
------
time go get github.com/blevesearch/bleve/...

real    0m42.638s
user    0m25.183s
sys     0m4.910s


the larger kagome variant also used around 2-3GB of RAM when go getting
it.

So the ipadic-only variant seems to take significantly less time to get
but I can't make the call on this.


Cheers,

Silvan

**Comment by mschoch** [comment]
Can someone comment on the appropriateness on using one dictionary vs the other?

If we moved it out to the blevex repo (which we still might do anyway) would some users still want/need access to both dictionaries?

**Comment by dc0d** [comment]
Any updates on this? (It really does not provide a pleasant experience).

**Comment by mschoch** [comment]
I've merged the PR switching us to the smaller version of Kagome, and also moved the package into the blevex repo.  I've posted about the change here:

https://groups.google.com/d/msg/bleve/XVGngMRsW7w/D7bmFVFMBwAJ

In short, my decision to move it to blevex is in anticipation that we will be adding support for other languages that also include large dictionaries.  I think this will improve the overall usability for bleve users and still make the language specific analyzers available.

I'm leaving this ticket open for now, as we need to improve documentation around the blevex package so that users that want to use it can easily do so.
